### PR TITLE
Amls 4272 - Supervision Section - page split

### PR DESCRIPTION
**Pull request by slavvco**
First question in supervision flow has beed divided into four.

Implementation required to extract new models, controllers and views from existing one. Apart from that there are changes in summary view for this section. Additionally json readers and writers in AMLS backend needed to be updated. 

## Related / Dependant PRs?

There's dependency with AMLS-4272 in backend, therefore:
**DON'T MERGE** without merging backend

## How Has This Been Tested?

- manually
- unit tests

## Checklist

- [x] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [x] Unit tests have full coverage.
- [x] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [ ] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

**Comment by slavvco** [comment]
I've made some updates as per requests from our QA and Content Designer. Added few unit tests and missing date validation. That's why commits above.

File: app/controllers/supervision/AnotherBodyController.scala [hunk 1]
```diff
@@ -47,16 +50,47 @@ class AnotherBodyController @Inject() (val dataCacheConnector: DataCacheConnecto
...
+    import utils.ControllerHelper.{anotherBodyComplete, isAnotherBodyComplete, isAnotherBodyYes}
+
+    val anotherBody = anotherBodyComplete(cache)
+
+    if (isAnotherBodyYes(anotherBody)) {
+      (edit, isAnotherBodyComplete(anotherBody)) match {
+        case (true, false) => Redirect(routes.SupervisionStartController.get())
+        case (false, false) => Redirect(routes.SupervisionStartController.get())
+        case (_, true) => Redirect(routes.SummaryController.get())
+      }
```
Comment (ajchilts) [inline app/controllers/supervision/AnotherBodyController.scala]: Can this be simplified?

I.e.

```
case (_, true) => Redirect(routes.SummaryController.get())
case (_, _) => Redirect(routes.SupervisionStartController.get())
```

File: app/controllers/supervision/SupervisionEndReasonsController.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    def supervisionComplete = cache.getEntry[Supervision](Supervision.key).get.isComplete
+
+    if (isAnotherBodyYes(anotherBody)) {
+      (edit, isAnotherBodyComplete(anotherBody)) match {
+        case (true, false) => Redirect(routes.ProfessionalBodyMemberController.get())
+        case (false, false) => Redirect(routes.ProfessionalBodyMemberController.get())
+        case (false, true) if supervisionComplete => Redirect(routes.SummaryController.get())
+        case (true, true) => Redirect(routes.SummaryController.get())
+        case (false, true) if !supervisionComplete => Redirect(routes.ProfessionalBodyMemberController.get())
+      }
```
Comment (ajchilts) [inline app/controllers/supervision/SupervisionEndReasonsController.scala]: Could the above be simplified?

File: app/controllers/supervision/AnotherBodyController.scala (hunk 1 above)
Comment (slavvco) [reply app/controllers/supervision/AnotherBodyController.scala]: done

File: app/controllers/supervision/SupervisionEndReasonsController.scala (hunk 2 above)
Comment (slavvco) [reply app/controllers/supervision/SupervisionEndReasonsController.scala]: done

**Comment by slavvco** [comment]
Tested that locally to confirm it works with old data format and that translation works fine for users with data stored in old one. Submissions/amendments are always converted to original schema.
Tests were performed to cover:
- receiving data through API,
- processing that,
- writing / reading to / from mongo 
- submitting data through API

**Comment by ajchilts** [comment]
@slavvco - Some merge conflicts to resolve and then I am happy! Well done as these page splits are involved. 👍

**Comment by slavvco** [comment]
Ok, done :)

File: app/controllers/supervision/AnotherBodyController.scala [hunk 3]
```diff
@@ -47,16 +50,40 @@ class AnotherBodyController @Inject() (val dataCacheConnector: DataCacheConnecto
...
+    supervision.anotherBody(updatedAnotherBody).copy(hasAccepted = true)
+  }
+
+  private def redirectTo(edit: Boolean, cache: CacheMap)(implicit authContext: AuthContext, headerCarrier: HeaderCarrier) = {
+
+    import utils.ControllerHelper.{anotherBodyComplete, isAnotherBodyYes}
+
+    val anotherBody = anotherBodyComplete(cache)
+
+    def supervisionComplete = cache.getEntry[Supervision](Supervision.key).get.isComplete
```
Comment (ajchilts) [inline app/controllers/supervision/AnotherBodyController.scala]: @slavvco - This get could be dangerous. Can we be sure we have a supervision cache entry here? Do we need to hand where we do not?

Comment (slavvco) [reply app/controllers/supervision/AnotherBodyController.scala]: Ok, done here and in three other places.

**Pull request merged**